validation.go

```diff
@@ -149,3 +190,71 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	}
 	return out
 }
+
+// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// consists of a combination of errors from the configurations.
+func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+	multiErr := assignParamDefaults(params, config)
+
+	for pKey, param := range params {
+		if config[pKey] != "" {
+			multiErr = multierr.Append(multiErr, validateParamType(param, pKey, config[pKey]))
+		}
+		multiErr = multierr.Append(multiErr, validateParamValue(pKey, config[pKey], param))
+	}
+	return multiErr
+}
+
+// validateParamValue validates that a configuration value matches all the validations required for the parameter.
+func validateParamValue(key string, value string, param Parameter) error {
+	var multiErr error
+	var err error
+
+	for _, v := range param.Validations {
+		err = v.validate(value)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("error validating %q: %w", key, err))
+		}
+	}
+	return multiErr
+}
+
+// assignParamDefaults fills any empty configuration with its assigned default value
+// returns an error if a parameter is not recognized.
+func assignParamDefaults(params map[string]Parameter, config map[string]string) error {
+	var multiErr error
+	for key, val := range config {
+		if _, ok := params[key]; !ok {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("unrecognized parameter %q", key))
+		}
+		if val == "" {
+			config[key] = params[key].Default
+		}
+	}
+	return multiErr
+}
+
+// validateParamType validates that a parameter value is parsable to its assigned type.
+func validateParamType(p Parameter, key string, value string) error {
+	switch p.Type {
+	case ParameterTypeNumber:
+		// a number should be parsable into float
+		_, err := strconv.ParseFloat(value, 64)
+		if err != nil {
+			return fmt.Errorf("%q: %q value is not a number: %w", key, value, ErrInvalidParamType)
+		}
+	case ParameterTypeDuration:
+		_, err := time.ParseDuration(value)
+		if err != nil {
+			return fmt.Errorf("%q: %q value is not a duration: %w", key, value, ErrInvalidParamType)
+		}
+	case ParameterTypeBool:
+		_, err := strconv.ParseBool(value)
+		if err != nil {
+			return fmt.Errorf("%q: %q value is not a boolean: %w", key, value, ErrInvalidParamType)
+		}
+		// type ParameterTypeFile and ParameterTypeString don't need type validations, since a file is an array of bytes,
```

Comment:
If ParameterTypeFile represents a file, which is an array of bytes, and since all configuration values are string, how exactly should a file be encoded as a string?

Reply:
I meant that an array of bytes is basically a string, so we can't validate that it's a file, it's just a string that doesn't make sense. so the UI would upload a file and convert it to string.
The whole file parameter is still not implemented in the UI and this would be used in the future.

Reply:
What I meant is that the string is an encoding of the input file. It may be a Base64-encoded version of the file for example. But since we don't support it now, I don't think it makes sense to "assign" meaning to it yet.